internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -98,77 +98,98 @@ func (md *machineDeployment) deployMachinesApp(ctx context.Context) error {
 	processGroupMachineDiff := md.resolveProcessGroupChanges()
 	md.warnAboutProcessGroupChanges(ctx, processGroupMachineDiff)
 
-	if len(processGroupMachineDiff.machinesToRemove) > 0 {
-		// Destroy machines that don't fit the current process groups
-		if err := md.machineSet.RemoveMachines(ctx, processGroupMachineDiff.machinesToRemove); err != nil {
-			return err
+	// TODO: validate _somewhere_ no mounts/volumes are used.
+	if md.strategy == "canary" {
```

Comment:
Our previous chat left me thinking that canaries are a waste of time on initial deploy (see `md.isFirstDeploy` bool)

Reply:
Ah, nice! Will incorporate this into the deploy.